front/src/applications/operationalStudies/views/SimulationResults.tsx

```diff
@@ -106,6 +116,30 @@ const SimulationResults = ({
 
   const conflictZones = useProjectedConflicts(infraId, conflicts, projectionData?.path);
 
+  const selectedTrainSummary = useMemo(
+    () => trainScheduleSummaries?.find((train) => train.id === selectedTrainSchedule?.id),
+    [trainScheduleSummaries, selectedTrainSchedule]
+  );
+
+  const handleTrainDrag = async (
+    draggedTrain: TrainSpaceTimeData,
```

Comment:
Seems we use only the dragged train id here. Maybe we could use only that in this function ?

```suggestion
    draggedTrainId: number,
```

Reply:
I made the change